src/rgw/rgw_common.cc

```diff
@@ -987,7 +987,7 @@ bool verify_object_permission(struct req_state * const s,
   }
 
   bool ret = object_acl->verify_permission(*s->auth_identity, s->perm_mask, perm);
-  if (ret) {
+  if (ret && s->cct->_conf->rgw_swift_verify_object_acl) {
```

Comment:
i think we should only check `rgw_swift_verify_object_acl` if we're processing a swift request (based on `prot_flags`). and instead of ignoring the return value of `object_acl->verify_permission()`, we should be able to avoid calling it in the first place. something like this?
```c++
  // allow configuration to skip object acl checks for swift requests
  bool verify_object_acl = !(s->prot_flags & RGW_REST_SWIFT) ||
                            s->cct->_conf->rgw_swift_verify_object_acl;

  if (verify_object_acl && object_acl->verify_permission(*s->auth_identity, s->perm_mask, perm)) {
    return true;
  }
```